spec/requests/application_controller_spec.rb

```diff
@@ -1520,4 +1520,20 @@ def preloaded_json
       end
     end
   end
+
+  describe "#set_current_user_for_logs" do
+    fab!(:admin)
+
+    it "sets the X-Discourse-Route header to the controller name and action including namespace" do
+      sign_in(admin)
+
+      get "/admin/users/#{admin.id}.json"
+      expect(response.status).to eq(200)
+      expect(response.headers["X-Discourse-Route"]).to eq("admin::users/show")
```

Comment:
so more deeply nested routes will look like, e.g. `admin::config::flags/toggle`?

Reply:
Yes.

Reply:
I wanted to do (1), but for the sake of minimizing the odds of breaking stuff — e.g. something currently splits on `/` — I went with double-colons which is not a strange syntax to ruby. I will check our internal tooling and see if anything references this header.

Reply:
hm, how about just changing specifically the admin prefix? something like:

```rb
controller_path.gsub(/^admin\// "admin::")
```

Reply:
Why just the admin prefix specifically? If we keep the slash for other namespaces, then we'd have to switch to `#` for the separator between the controller name and action, otherwise the header would look like e.g. `chat/api/current_user_channels/index`.

Reply:
ah, I thought admin was the only namespace…

taking a step back - do we need a different separator for the action? or are `admin/users/show`, `users/show` and `chat/api/current_user_channels/index` fine?

Reply:
Hmm that's a good point — I can't actually think of any reasons to have a different separator for the action. The names of the action and controller (with namespace) is what really matters, so as long as they're included, I think we're good. Pushing an update now...